controllers/provisioners/eks/helpers.go

```diff
@@ -1173,3 +1173,40 @@ func (ctx *EksInstanceGroupContext) GetDesiredMixedInstancesPolicy(name string)
 
 	return policy
 }
+
+func FilterSupportedArch(architectures []string) string {
+	for _, a := range architectures {
+		for _, supportedArch := range SupportedArchitectures {
+			result := a == supportedArch
+			if result {
+				return supportedArch
+			}
+		}
+	}
+	return ""
+}
+
+func (ctx *EksInstanceGroupContext) GetEksLatestAmi() (string, error) {
+	var (
+		instanceGroup = ctx.GetInstanceGroup()
+		state         = ctx.GetDiscoveredState()
+		configuration = instanceGroup.GetEKSConfiguration()
+	)
+	clusterVersion := state.GetClusterVersion()
+	annotations := instanceGroup.GetAnnotations()
+
+	var OSFamily string
+	if kubeprovider.HasAnnotation(annotations, OsFamilyAnnotation) {
```

Comment:
previous version, now `HasAnnotationWithValue` caused some unexpected logic problems; I've added a new method to just detect the presence of a key in the annotation